hostapd/hostapd_cli_cmds.c

```diff
@@ -0,0 +1,1374 @@
+                       "argument, STA address, is required.\n");
+                return -1;
+        }
+        if (argc > 1)
+                os_snprintf(buf, sizeof(buf), "DISASSOCIATE %s %s",
+                            argv[0], argv[1]);
+        else
+                os_snprintf(buf, sizeof(buf), "DISASSOCIATE %s", argv[0]);
+        return hostapd_ctrl_command(ctrl, buf);
+}
+
+
+static int hostapd_cli_cmd_deauthenticate(struct wpa_ctrl *ctrl, int argc,
+                                          char *argv[])
+{
+        char buf[64];
+        if (argc < 1) {
+                printf("Invalid 'deauthenticate' command - exactly one "
+                       "argument, STA address, is required.\n");
+                return -1;
+        }
+        if (argc > 1)
+                os_snprintf(buf, sizeof(buf), "DEAUTHENTICATE %s %s",
+                            argv[0], argv[1]);
+        else
+                os_snprintf(buf, sizeof(buf), "DEAUTHENTICATE %s", argv[0]);
+        return hostapd_ctrl_command(ctrl, buf);
+}
+
+
+static int hostapd_cli_cmd_status(struct wpa_ctrl *ctrl, int argc, char *argv[])
+{
+	if (argc > 0 && os_strcmp(argv[0], "driver") == 0)
+		return hostapd_ctrl_command(ctrl, "STATUS-DRIVER");
+	return hostapd_ctrl_command(ctrl, "STATUS");
+}
+
+
+static int hostapd_cli_cmd_enable(struct wpa_ctrl *ctrl, int argc,
+                                  char *argv[])
+{
+        return hostapd_ctrl_command(ctrl, "ENABLE");
+}
+
+
+static int hostapd_cli_cmd_reload(struct wpa_ctrl *ctrl, int argc,
+                                  char *argv[])
+{
+        return hostapd_ctrl_command(ctrl, "RELOAD");
+}
+
+
+static int hostapd_cli_cmd_disable(struct wpa_ctrl *ctrl, int argc,
+                                   char *argv[])
+{
+        return hostapd_ctrl_command(ctrl, "DISABLE");
+}
+
+
+static int hostapd_cli_cmd_update_beacon(struct wpa_ctrl *ctrl, int argc,
+                                         char *argv[])
+{
+        return hostapd_ctrl_command(ctrl, "UPDATE_BEACON");
+}
+
+
+static int hostapd_cli_cmd_sta(struct wpa_ctrl *ctrl, int argc, char *argv[])
+{
+	char buf[64];
+	if (argc < 1) {
+		printf("Invalid 'sta' command - at least one argument, STA "
+		       "address, is required.\n");
+		return -1;
+	}
+	if (argc > 1)
+		snprintf(buf, sizeof(buf), "STA %s %s", argv[0], argv[1]);
+	else
+		snprintf(buf, sizeof(buf), "STA %s", argv[0]);
+	return hostapd_ctrl_command(ctrl, buf);
+}
+
+
+static char ** hostapd_complete_stations(const char *str, int pos)
+{
+	int arg = get_cmd_arg_num(str, pos);
+	char **res = NULL;
+
+	switch (arg) {
+	case 1:
+		res = cli_txt_list_array(&stations);
+		break;
+	}
+
+	return res;
+}
+
+
+static int hostapd_cli_cmd_new_sta(struct wpa_ctrl *ctrl, int argc,
+				   char *argv[])
+{
+	char buf[64];
+	if (argc != 1) {
+		printf("Invalid 'new_sta' command - exactly one argument, STA "
+		       "address, is required.\n");
+		return -1;
+	}
+	snprintf(buf, sizeof(buf), "NEW_STA %s", argv[0]);
```

Comment:
Perhaps we should have here `os_snprintf()` like in other functions.

Reply:
Updated.